code/game/machinery/machinery.dm

```diff
@@ -192,6 +207,8 @@ Class Procs:
 		use_power(idle_power_usage,power_channel, 1)
 	else if(use_power >= ACTIVE_POWER_USE)
 		use_power(active_power_usage,power_channel, 1)
+	if(prob(MACHINE_FLICKER_CHANCE))
```

Comment:
I'm really not sure about the idea of machines randomly flickering (ie: doing prob() calls regularly).

Reply:
From a technical, or design perspective?

Reply:
Technical I'd imagine. It would be cool if you could profile the performance cost of this, i.e run a 30 minute test on master vs 30m with your changes. I don't think it will be *that* much of a cost, but worth checking out.

Reply:
I profiled it before I posted the PR, which is why I asked the question. Didn't save the results, so if it would make folks feel better I can profile it again when I have a few moments spare.

Reply:
Profiled the results of a run over 30 minutes each.

With the code enabled:
![profile1-code_is_in](https://user-images.githubusercontent.com/16298462/95907572-bdaeaa80-0d69-11eb-812e-52900bc49064.png)

With the code disabled:
![profile2-code_is_out](https://user-images.githubusercontent.com/16298462/95907600-c69f7c00-0d69-11eb-9247-8220bfb73314.png)

Variance in the number of calls can be explained by space ruins and away missions or slight changes in how I timed the runs, but it seems pretty clear this doesn't impact the proc in any major way.

Reply:
Yep, this is negligible, and not really a concern. And all the other stuff (computers flickering etc.) happens so infrequently I don't think that will be a problem either.